packages/devextreme/js/__internal/ui/m_load_indicator.ts

```diff
@@ -113,25 +137,47 @@ class LoadIndicator extends Widget<LoadIndicatorProperties> {
     }
   }
 
-  _renderAnimationMarkup(): void {
-    const animatingSegmentInner = this.option('_animatingSegmentInner');
+  _getSegmentParams(): SegmentParams {
+    const {
+      _animationType: animationType,
+      _animatingSegmentCount: animatingSegmentCount,
+      _animatingSegmentInner: animatingSegmentInner,
+    } = this.option();
+
+    const segmentParams = {
+      [AnimationType.Circle]: {
+        segmentCount: animatingSegmentCount ?? 0,
+        segmentInner: Boolean(animatingSegmentInner),
+      },
+    };
+
+    return segmentParams[animationType];
+  }
 
+  _renderAnimationMarkup(): void {
     this._$indicator = $('<div>').addClass(LOADINDICATOR_ICON_CLASS);
     this._$content.append(this._$indicator);
 
-    // Indicator markup
-    // @ts-expect-error ts-error
-    for (let i = this.option('_animatingSegmentCount'); i >= 0; --i) {
+    const params = this._getSegmentParams();
+
+    this._renderSegments(params);
+  }
+
+  _renderSegments(params: SegmentParams): void {
+    const { segmentCount, segmentInner } = params;
+
+    for (let i = segmentCount; i >= 0; i -= 1) {
       const $segment = $('<div>')
         .addClass(LOADINDICATOR_SEGMENT_CLASS)
-        // eslint-disable-next-line @typescript-eslint/restrict-plus-operands, @typescript-eslint/no-base-to-string
-        .addClass(LOADINDICATOR_SEGMENT_CLASS + i);
+        .addClass(`${LOADINDICATOR_SEGMENT_CLASS}${i}`);
+
+      if (segmentInner) {
+        const $segmentInner = $('<div>').addClass(LOADINDICATOR_SEGMENT_INNER_CLASS);
 
-      if (animatingSegmentInner) {
-        $segment.append($('<div>').addClass(LOADINDICATOR_SEGMENT_INNER_CLASS));
+        $segment.append($segmentInner);
       }
 
-      this._$indicator.append($segment);
+      this._$indicator?.append($segment);
```

Comment:
Is optional chaining intentional here?

Reply:
Yes, since TS in this case cannot determine that this._$indicator currently exists. We can either explicitly specify that this._$indicator already exists (but this is forbidden by ESLint), or we can add a check. In this case, I think adding a check is safer.